### PR TITLE
dns/dnscrypt-proxy: fix logging

### LINKED ISSUE #2440: DNSCrypt-Proxy logs no longer available in WebGUI
After upgrading to 21.1.7 (and the 21.1.7_1 hotfix), sadly logs are no longer showing in the WebGUI for the DNSCrypt-Proxy (1.8_1)

Service is still operational and still working, just logs are absent from WebGUI.

OPNsense 21.1.7_1-amd64
FreeBSD 12.1-RELEASE-p18-HBSD
OpenSSL 1.1.1k  25 Mar 2021
Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz (2 cores)

### PR DESCRIPTION
**Pull request by mimugmail**
Fix logging due to Phalcon4 update

Closes #2440

### CODE REVIEW

**Comment by L1ghtn1ng** [comment]
@fichtner is it possible to get this as a hotfix?

**Comment by fichtner** [comment]
It's not directly related to operational issues so I would say this has to wait for 21.1.9.

**Comment by L1ghtn1ng** [comment]
This missed 21.1.9 can we get this in 21.7 please?

**Comment by fichtner** [comment]
21.7 was built last week. Just apply the patch manually for the time being.

File: dns/dnscrypt-proxy/src/opnsense/scripts/OPNsense/Dnscryptproxy/setup.sh [hunk 1]
```diff
@@ -2,3 +2,4 @@
 
 mkdir -p /var/log/dnscrypt-proxy
 chown -R _dnscrypt-proxy:_dnscrypt-proxy /var/log/dnscrypt-proxy
+ln -s /var/log/dnscrypt-proxy /var/log/dnscryptproxy
```
Comment (fichtner) [inline dns/dnscrypt-proxy/src/opnsense/scripts/OPNsense/Dnscryptproxy/setup.sh:5]:
```suggestion
(cd /var/log && ln -s dnscrypt-proxy dnscryptproxy)
```

**Comment by fichtner** [comment]
Merged, thanks!

**Comment by Malaach** [comment]
Seems like logging has stopped working again - I just upgraded to OPNsense 22.1 (version 1.10 of the plugin) and am missing the logs. I can't tell however if that bug was re-introduced with 22.1 or earlier.

Tried deactivating and reactivating option "Enable query logs" which didn't improve the situation.

**Pull request merged**